src/typescript_proto_library.bzl

```diff
@@ -230,20 +304,54 @@ def _typescript_proto_library_impl(ctx):
         ],
     )
 
-typescript_proto_library = rule(
+def typescript_proto_library(name, proto):
+    typescript_proto_build(
+        name = name,
+        proto = proto,
+
+        generate = "base",
+    )
+
+typescript_proto_build = rule(
     attrs = {
         "proto": attr.label(
+            doc = "the protobuf file that should be processed",
             mandatory = True,
             allow_single_file = True,
             providers = [ProtoInfo],
             aspects = [typescript_proto_library_aspect],
         ),
+        "generate": attr.string(
+            doc = "which format to output, with 'base' being the protobuf messages",
+            mandatory = False,
+            default = "base",
+            values = [
+                "base",
+                "grpc-node",
+                "grpc-web",
+            ],
+        ),
+        "mode": attr.string(
+            doc = "when using grpc-node, either the grpc library can be referenced (default) or grpc-js",
+            mandatory = False,
+            default = "",
+            values = [
+                "",
+                "grpc-js",
+            ],
+        ),
```

Comment:
Optional: I'm not very familiar with grpc-node, but would it make sense to have two types, `grpc-node-with-grpc` or `grpc-node-with-grpc-js` instead of mode?

Alternatively, we could name this field grpc_node_mode since it's only used for grpc_node.

Reply:
In the `grpc-node` ecosystem, there is `grpc` and `grpc-js`. The generated files from `ts-protoc-gen` are nearly identical except that the require statements reference one or the other. Renaming it makes a lot of sense 👍 .